docker/docker_client.go

```diff
@@ -187,45 +187,55 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool)
 	}, nil
 }
 
+type authScope struct {
+	remoteName string
+	actions    string
+}
+
+type requestOptions struct {
+	scope   authScope
+	headers map[string][]string
+	stream  io.Reader
+}
```

Comment:
I don’t really feel strongly about introducing this… having named optional parameters is nice, I suppose. Though if we end up having 2 or 3, it is not a big deal.